### PR TITLE
alps: new package

### PR DESCRIPTION
**Pull request by Sinan81**
Latest source file builds with newer python. Hence re-add the removed ALPS package.

### CODE REVIEW

**Comment by Sinan81** [comment]
@spackbot fix style

**Comment by Sinan81** [comment]
FYI @andredepinhovieira @ketsubouchi

**Comment by Sinan81** [comment]
@adamjstewart this package was removed by you due to obsolete python dependency. In this PR I've recreated it from scratch. Is this ok, or else is there a better way to restore a removed package?

File: var/spack/repos/builtin/packages/alps/package.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    depends_on("python", type=("build", "link", "run"))
+    depends_on("py-numpy", type=("build", "run"))
+    depends_on("py-scipy", type=("build", "run"))
+    depends_on("py-matplotlib", type=("build", "run"))
+
+    extends("python")
+
+    def url_for_version(self, version):
+        if str(version) == "2.3.0":
+            url = "https://alps.comp-phys.org/static/software/releases/alps-2.3.0-src.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/alps/package.py]: If this is no longer supported, we can simplify this function to work for all versions.

File: var/spack/repos/builtin/packages/alps/package.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        elif str(version) == "20220304_r7871":
+            url = "http://exa.phys.s.u-tokyo.ac.jp/archive/MateriApps/src/alps_20220304~r7871.orig.tar.gz"
+        return url
+
+    def cmake_args(self):
+        args = []
+        args.append("Boost_ROOT_DIR=" + self.spec["boost"].prefix)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/alps/package.py]: Should this have a `-D`? I usually use `self.define(...)` for these.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by adamjstewart** [comment]
Ping me when it's ready to review.

**Comment by Sinan81** [comment]
For some reason, the package build is failing. I built it many times before. i need to sort that out.

**Comment by Sinan81** [comment]
will push a new MR. closing

**Pull request closed without merging**